app/views/editor/level/modals/SaveBranchModal.coffee

```diff
@@ -0,0 +1,185 @@
+    @selectedBranchDeltaViews = []
+    @renderSelectors('#selected-branch-col')
+    changeEls = @$el.find('#selected-branch-col .component-changes-stub')
+    for changeEl in changeEls
+      componentId = $(changeEl).data('component-id')
+      component = @selectedBranch.components.get(componentId)
+      targetComponent = @components.find((c) -> c.get('original') is component.get('original') and c.get('version').isLatestMajor)
+      preBranchSave = component.clone()
+      preBranchSave.markToRevert()
+      componentDiff = targetComponent.clone()
+      preBranchSave.set(componentDiff.attributes)
+      @selectedBranchDeltaViews.push(@insertDeltaView(preBranchSave, changeEl))
+
+    changeEls = @$el.find('#selected-branch-col .system-changes-stub')
+    for changeEl in changeEls
+      systemId = $(changeEl).data('system-id')
+      system = @selectedBranch.systems.get(systemId)
+      targetSystem = @systems.find((c) -> c.get('original') is system.get('original') and c.get('version').isLatestMajor)
+      preBranchSave = system.clone()
+      preBranchSave.markToRevert()
+      systemDiff = targetSystem.clone()
+      preBranchSave.set(systemDiff.attributes)
+      @selectedBranchDeltaViews.push(@insertDeltaView(preBranchSave, changeEl))
+
+  onClickBranch: (e) ->
+    $(e.currentTarget).closest('.list-group').find('.active').removeClass('active')
+    $(e.currentTarget).addClass('active')
+    branchCid = $(e.currentTarget).data('branch-cid')
+    @selectedBranch = if branchCid then @branches.get(branchCid) else null
+    @renderSelectedBranch()
+
+  onClickStashBranchButton: (e) ->
+    @deleteSavedChanges(e, {deleteSavedChanges: true})
+
+  onClickSaveBranchButton: (e) ->
+    @deleteSavedChanges(e, {deleteSavedChanges: false})
+    
+  deleteSavedChanges: (e, {deleteSavedChanges}) ->
+    if @selectedBranch
+      branch = @selectedBranch
+    else
+      name = @$('#new-branch-name-input').val()
+      if not name
+        return noty text: 'Name required', layout: 'topCenter', type: 'error', killer: false
+      slug = _.string.slugify(name)
+      if @branches.findWhere({slug})
+        return noty text: 'Name taken', layout: 'topCenter', type: 'error', killer: false
+      branch = new Branch({name})
+    
+    patches = []
+    toRevert = []
+    selectedComponents = _.map(@$('.component-checkbox:checked'), (checkbox) => @componentsWithChanges.get($(checkbox).data('component-id')))
+    for component in selectedComponents
+      patches.push(component.makePatch().toJSON())
+      toRevert.push(component)
+    
+    selectedSystems = _.map(@$('.system-checkbox:checked'), (checkbox) => @systemsWithChanges.get($(checkbox).data('system-id')))
+    for system in selectedSystems
+      patches.push(system.makePatch().toJSON())
+      toRevert.push(system)
+    branch.set({patches})
+    jqxhr = branch.save()
+    button = $(e.currentTarget)
+    if not jqxhr
+      return button.text('Save Failed')
```

Comment:
Does `.save()` ever return a falsey value? Did you mean to wait for the return value?